src/ReflectionProperty.php

```diff
@@ -190,43 +176,53 @@ public function getValue($object = null)
     }
 
     /**
-     * @inheritDoc
+     * {@inheritDoc}
      */
-    public function isDefault()
+    public function isDefault(): bool
     {
-        // TRUE if the property was declared at compile-time
+        $this->initializeInternalReflection();
 
-        return true;
+        return parent::isDefault();
     }
 
     /**
      * {@inheritDoc}
      */
-    public function isPrivate()
+    public function getDefaultValue(): mixed
+    {
+        $this->initializeInternalReflection();
```

Comment:
Sorry, this won't work properly for AOP needs because of autoloader process restrictions, `goaop` **must not** use core reflection all the time, otherwise AOP won't work at all ((